### PR TITLE
Ensure active button styles are not applied to disabled buttons

### LINKED ISSUE #23353: Active state of disabled button is  triggering when <i> or <b> inside a button is clicked.
- Operating system: Windows 7 SP1.
- Browser: Chrome v 60.0.3112.90, (64 bit)
- https://jsbin.com/vewexuviqi/1/edit?html,css,js,output  - first 2 buttons is bugged, last one isn't.

`<i>123</i>, <b>123</b>` tags inside of `<button type="button" class="btn btn-lg btn-outline-dark" disabled></button>` tag causes triggering active state of button when button is clicked. 
(`<a>123</a>` and maybe other tags causes this behavior too).
In Firefox 54.0.1 (32-bit), Microsoft Edge 38.14393.1066.0 and IE 11.0.9600.18762 all disabled buttons is triggering active state regardless of `<i>` or `<b>` tags.

### PR DESCRIPTION
**Pull request by prateekgoel**
I have added the styles same as `.disabled` to `.disabled:active`.
This should fix #23353.

Let me know if any other changes are required.

### CODE REVIEW

**Comment by patrickhlauke** [comment]
Haven't had a chance to test it, but one question that pops into my head: are there situations where a `.disabled` element receives the programmatically set `.active` class? I *think* our scripts should check for that/prevent that.

If `.active` isn't set to things that are `.disabled`, I'd slim down the rules to only cover `:active` pseudo

**Comment by prateekgoel** [comment]
I am not sure if any disabled element receives `.active` class. 
If someone can confirm, it would be great. We can eliminate some rules then.
CC @mdo

File: scss/_buttons.scss [hunk 1]
```diff
@@ -27,7 +27,11 @@
 
   // Disabled comes first so active can properly restyle
   &.disabled,
-  &:disabled {
+  &:disabled,
+  &.disabled:active,
+  &:disabled:active,
+  &.disabled.active,
```
[the remark this answers is not in the document]
Comment (prateekgoel) [reply scss/_buttons.scss]: Hi @andresgalante 
These to lines are not exactly same. First one applies to active state of button having disabled class and the second one applies to active state of disabled button with disabled attribute.

[the remark this answers is not in the document]
Comment (andresgalante) [reply scss/_buttons.scss]: right! sorry I miss read the `:` and the `.`
Thanks for the clarification

**Comment by patrickhlauke** [comment]
an alternative, more terse way (instead of redefining `:active`/`.active` for when the button is disabled) would be to make the current active selectors more specific, e.g. `&:not([disabled]):not(.disabled):active, &:not([disabled]):not(.disabled).active` - or would that be too cryptic?

**Comment by prateekgoel** [comment]
Sorry @patrickhlauke , didn't get you.
Using `:not` will target the buttons that are not disabled. We need to target the disabled buttons, right?

**Comment by patrickhlauke** [comment]
@prateekgoel what i mean is: instead of redefining new styles for disabled+active scenario, you could likely save yourself all the redefinition by instead making the current active styles for buttons only target buttons that aren't disabled.

**Comment by patrickhlauke** [comment]
i.e. not adding new style blocks, and instead changing

```
&:active,
&.active
```

to

```
&:not([disabled]):not(.disabled):active,
&:not([disabled]):not(.disabled).active
```

**Comment by prateekgoel** [comment]
Yes, got your point. Seems a nice idea.

**Comment by prateekgoel** [comment]
Thanks for the suggestion. 😃

**Comment by prateekgoel** [comment]
@patrickhlauke Did the changes.

**Review by patrickhlauke — APPROVED** [review]

**Comment by patrickhlauke** [comment]
Sorry for the delay...but finally able to build locally and test, and it works well (even in IE11) :)

**Pull request merged**